### PR TITLE
Fixed issue with geotiff access from protected s3 bucket

### PR DESCRIPTION
**Pull request by TejasMorbagal**
[Description of PR]

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [ ] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## Changes in 1.1.2
```
Comment (TonioF) [inline CHANGES.md:1]:
```suggestion
## Changes in 1.1.2 (in development)
```

File: xcube/core/store/fs/impl/dataset.py [hunk 2]
```diff
@@ -376,13 +377,18 @@ def open_data(self,
...
     def create_env_session(cls, fs):
         if isinstance(fs, s3fs.S3FileSystem):
-            return rasterio.env.Env(aws_secret_access_key=fs.token,
-                                    aws_access_key_id=fs.key,
-                                    aws_session_token=fs.token,
-                                    region_name=fs.client_kwargs.get(
-                                        'region_name',
-                                        'eu-central-1'
-                                    ),
+            # Create an AWSSession object as Passing abstract session keyword
```
Comment (thomasstorm) [inline xcube/core/store/fs/impl/dataset.py:380]: Please remove comment, The code is pretty clear and stands for itself.

If you insist on keeping a comment here, please rephrase as I think it's hard to understand in its current state.

**Review by thomasstorm — APPROVED** [review]
Approved, but please consider my comment.

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+
+### Fixes
+
+* Fixed issue with geotiff access from protected s3 bucket
```
Comment (TonioF) [inline CHANGES.md:5]:
```suggestion
* Fixed issue where geotiff access from a protected s3 bucket was denied (#864)
```

**Review by TonioF — CHANGES_REQUESTED** [review]
See my comments (all small stuff).

**Review by AliceBalfanz — APPROVED** [review]
I approve under the condition, that Tonios comments are addressed :)

File: xcube/core/store/fs/impl/dataset.py (hunk 2 above)
Comment (TejasMorbagal) [reply xcube/core/store/fs/impl/dataset.py]: done

File: CHANGES.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+## Changes in 1.1.2 (in development)
+
+### Fixes
+
+* Fixed issue where geotiff access from a protected s3 bucket was denied (#864)
```
Comment (pont-us) [inline CHANGES.md:5]:
```suggestion
* Fixed issue where geotiff access from a protected s3 bucket was denied (#863)
```

**Review by pont-us — APPROVED** [review]
Approving, but please fix the typo I found in the changelog :)

**Pull request merged**